templates/expo-template-bare-minimum/ios/Podfile

```diff
@@ -13,15 +13,23 @@ install! 'cocoapods',
 
 prepare_react_native_project!
 
-flipper_config = ENV['NO_FLIPPER'] == "1" || ENV['CI'] ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
-
 target 'HelloWorld' do
   use_expo_modules!
   config = use_native_modules!
 
   use_frameworks! :linkage => podfile_properties['ios.useFrameworks'].to_sym if podfile_properties['ios.useFrameworks']
   use_frameworks! :linkage => ENV['USE_FRAMEWORKS'].to_sym if ENV['USE_FRAMEWORKS']
 
+  # Flipper support in expo via build properties plugin
+  expo_flipper = FlipperConfiguration.disabled
+  if ENV['NO_FLIPPER'] != "1" && defined?(podfile_properties['ios.flipper']) then
+    if podfile_properties['ios.flipper'] == 'true' then
+      expo_flipper = FlipperConfiguration.enabled(["Debug", "Release"])
+    elsif podfile_properties['ios.flipper'] != 'false' then
+      expo_flipper = FlipperConfiguration.enabled(["Debug", "Release"], { 'Flipper' => podfile_properties['ios.flipper'] })
+    end
+  end
+
```

Comment:
```suggestion

# Flipper support in expo via build properties plugin
expo_flipper = FlipperConfiguration.disabled
if ENV['NO_FLIPPER'] != "1" then
  if podfile_properties['ios.flipper'] == nil || podfile_properties['ios.flipper'] == 'false' then
    expo_flipper = FlipperConfiguration.disabled
    # Also setup NO_FLIPPER for 3rd party library like react-native-flipper to differentiate current mode
    ENV['NO_FLIPPER'] = '1'
  elsif podfile_properties['ios.flipper'] == 'true' then
    expo_flipper = FlipperConfiguration.enabled(["Debug"])
  else
    expo_flipper = FlipperConfiguration.enabled(["Debug"], { 'Flipper' => podfile_properties['ios.flipper'] })
  end
end

target 'HelloWorld' do
  use_expo_modules!
  config = use_native_modules!

  use_frameworks! :linkage => podfile_properties['ios.useFrameworks'].to_sym if podfile_properties['ios.useFrameworks']
  use_frameworks! :linkage => ENV['USE_FRAMEWORKS'].to_sym if ENV['USE_FRAMEWORKS']
```

after checking the build error on 0.71, i think the issue is coming from react-native-flipper if you add the dependency. since react-native-flipper autolinking does not respect release build type, even on release build, it will build source code which needs FlipperKit. if we don't have FlipperKit on release build, then it will have the build error. [the comments from 0.71](https://github.com/facebook/react-native/blob/9aa9ee79673b6ace86151f5543018f49dd06b774/template/ios/Podfile#L7-L15) makes sense to turn off react-native-flipper autolinking if we don't include FlipperKit.

i am proposing the change with three ideas, please let me know how do you think.
  - `FlipperConfiguration.enabled(["Debug"])`, to remove Release build variant
  - set `ENV['NO_FLIPPER'] = '1'` when `ios.flipper` podfile property is `false` or undefined. so that other library like react-native-flipper can differentiate the current flipper type.
  - move the code block with `ENV['NO_FLIPPER'] = '1'` out of Podfile target, so that the `ENV['NO_FLIPPER'] = '1'`  is setup before autolinking.

testing results:
| pod install command      | ios.flipper podfile property | Result: FlipperConfiguration.flipper_enabled | Result: FlipperConfiguration.versions | Result: ENV['NO_FLIPPER'] |
| ------------------------ | ---------------------------- | -------------------------------------------- | ------------------------------------- | ------------------------- |
| pod install              | undefined                    | false                                        | {}                                    | 1                         |
| pod install              | "true"                       | true                                         | {}                                    | undefined                 |
| pod install              | "0.174.0"                    | true                                         | {"Flipper"=>"0.174.0"}                | undefined                 |
| NO_FLIPPER=1 pod install | "true"                       | false                                        | {}                                    | 1                         |

Reply:
See https://github.com/expo/expo/pull/20890#issuecomment-1413222324 for resolution